src/map_io/map_allowed_building_types_packet.cc

```diff
@@ -74,10 +74,10 @@ void MapAllowedBuildingTypesPacket::read(FileSystem& fs,
 
 					bool allowed;
 					while (const char* const name = s.get_next_bool(nullptr, &allowed)) {
-						const DescriptionIndex index = tribe.building_index(name);
-						if (tribe.has_building(index)) {
+						try {
+							const DescriptionIndex index = tribe.safe_building_index(name);
 							player->allow_building_type(index, allowed);
-						} else {
+						} catch (const GameDataError&) {
```

Comment:
Instead of try-catch, use `building_index` rather than `safe_building_index` and compare the return value against `INVALID_INDEX`

Reply:
But then I have to expose the legacy lookup again, which I wanted to hide from the saveloading code to make it simpler.